mlflow/server/js/src/model-registry/components/SchemaTable.tsx

```diff
@@ -23,20 +26,92 @@ type Props = {
   };
 };
 
+function getTensorTypeRepr(tensorType: TensorSpec): string {
+  return `Tensor (dtype: ${tensorType['tensor-spec'].dtype}, shape: [${tensorType['tensor-spec'].shape}])`;
+}
+
+// return a formatted string representation of the column type
+function getColumnTypeRepr(columnType: ColumnType, indentationLevel: number): string {
+  const { type } = columnType;
+
+  const indentation = ' '.repeat(indentationLevel * INDENTATION_SPACES);
+  if (type === 'object') {
+    const propertyReprs = Object.keys(columnType.properties).map((propertyName) => {
+      const property = columnType.properties[propertyName];
+      const requiredRepr = property.required ? '' : ' (optional)';
+      const propertyRepr = getColumnTypeRepr(property, indentationLevel + 1);
+      const indentOffset = (indentationLevel + 1) * INDENTATION_SPACES;
+
+      return `${' '.repeat(indentOffset)}${propertyName}: ${
+        propertyRepr.slice(indentOffset) + requiredRepr
+      }`;
+    });
+
+    return `${indentation}{\n${propertyReprs.join(',\n')}\n${indentation}}`;
+  }
+
+  if (type === 'array') {
+    const indentOffset = indentationLevel * INDENTATION_SPACES;
+    const itemsTypeRepr = getColumnTypeRepr(columnType.items, indentationLevel).slice(indentOffset);
+    return `${indentation}Array(${itemsTypeRepr})`;
+  }
+
+  return `${indentation}${type}`;
+}
```

Comment:
/ai suggest ideas to simplify this function

Reply:
/ai suggest ideas to simplify this function

Reply:
@daniellok-db Never mind this comment. Just a test to see what GPT-4 is capable of.